aziotctl/src/internal/check/checks/read_key_pairs.rs

```diff
@@ -82,23 +85,54 @@ impl ReadKeyPairs {
                 })?;
                 Ok::<_, anyhow::Error>(key)
             };
+
             match key() {
                 Ok(key) => {
+                    let key_length = key.bits();
+
+                    match key.id() {
+                        openssl::pkey::Id::RSA => {
+                            if key_length < RSA_RECOMMENDED_MIN_BITS {
+                                warn_aggregated.push(format!(
+                                    "RSA key {} has length {} (min recommended: {})",
+                                    id, key_length, RSA_RECOMMENDED_MIN_BITS
+                                ));
+                            }
+                        }
+                        openssl::pkey::Id::EC => {
+                            if key_length != 256 {
```

Comment:
You should check the curve rather than the number of bits, because it could be another curve that's 256 bits.

```rust
key.ec_key().group().curve_name() == Some(openssl::nid::Nid::X9_62_PRIME256V1)
```

Reply:
Fixed